### ISSUE TITLE
Allow to always force facet refinement

### ISSUE THREAD

**Issue by benjamindulau** [algolia/instantsearch.js#1045]
It would be nice to "always" force a facet refinement without being able to clear it from the UI. Meaning the default "state" would be with the forced facet value so it's not considered "filtered" yet by the UI and thus the "clear all" widget wouldn't allow to clear this "forced" value.

Here is an use case: say I have some products each with a "brand" and a "color" attributes. I want a "global" search page with instant search which allows the user to control both brand & color facets refinements. But I also want a dedicated page for each brand and with only the color facet, and a dedicated page for each color and with only the brand facet.

In my case it's not an option to create an index for each possibility since in the real world I have much more attributes than just "brand" and "color" (without mentioning an index for each locale) and I want to have a dedicated page for each attribute (dynamically).

For now the only possibility I have with the library is the following:

```javascript
// For the "Apple" brand dedicated page for instance

const search = instantsearch({
    // ...
    searchParameters: {
    	disjunctiveFacets: ['brand']
    }
});

search.addWidget({
	init: function(args) {
    	args.helper.addDisjunctiveFacetRefinement('brand', 'Apple');
    }
});

search.addWidget(instantsearch.widgets.refinementList({
    container: '#color',
    attributeName: 'color',
    operator: 'or'
}));

search.start();
```

Is there some undocumented black magic solution I'm not aware of? Otherwise, would you consider such behavior in the lib? ;-)

Thanks.

**Comment by choomz** [comment]
Hello,

Great question. We also need this behaviour. @vvo does the lib can support this out of the box ? Or it needs some new feature ?

Thanks for answer :)

**Comment by pixelastic** [comment]
There is not easy way to do that in the current version. You can define default filters like you did, but they will still be removed with a `clearAll` and be displayed in the related refinement list.

This is an issue that pops up regularly (we have a longer discussion about it here: https://github.com/algolia/instantsearch.js/issues/740). Let me know if this is the same issue you're referring to so we can put all discussion on the same topic.

Long story short, we totally agree that this is needed, but we couldn't yet find a public API that would be satisfying. With more real-world example I'm sure we'll find something.

**Comment by vvo** [comment]
One way to solve this is to pass a list of parameters to not clear to the clearAll widget

**Comment by bobylito** [comment]
#addToFAQ

**Comment by pixelastic** [comment]
> One way to solve this is to pass a list of parameters to not clear to the clearAll widget

How should we do that? I don't see any related option in the `clearAll` widget.

**Comment by benjamindulau** [comment]
> Let me know if this is the same issue you're referring to so we can put all discussion on the same topic.

@pixelastic Yeah it's exactly the same issue as discussed in #740 

> One way to solve this is to pass a list of parameters to not clear to the clearAll widget

@vvo Is this already possible in the current version?

**Comment by vvo** [comment]
> @vvo Is this already possible in the current version?

Not yet no.

**Comment by vvo** [comment]
Closing in favor of #1063. clearAll/currentRefinedValues should support not clearing some attributes.
